### ISSUE TITLE
Specific inclusions/exclusions from store/restore

### ISSUE THREAD

**Issue by thesamovar** [brian-team/brian2#580]
One thing that people sometimes want to do is to store/restore all variables except for a couple that retain their values. It would be nice to have an easy way of supporting this in the syntax of store/restore rather than having them have to do it by hand. For example, you could ``store(exclude=[G.v, syn.w])``.

**Comment by mstimberg** [comment]
I agree that this could be a useful feature. I am a bit worried that this gives a mechanism to introduce inconsistencies, though. restore will write to internal variables that the user normally could not change directly (e.g. synapses), with this mechanism you could exclude such variables which would mess with the internals. Hmm, on the other hand maybe a simple solution would be to only allow the exclusion of variables that the user can change, i.e. only variables with `read_only == False`. This would exclude synapses, a `NeuronGroup`'s `N` variable, the time of a clock, etc.

**Comment by thesamovar** [comment]
I think read only variables would be fine.

**Comment by thesamovar** [comment]
I mean the opposite: writeable variables. :)
